src/factories/custom/Restricted_PIM_Factory_v1.sol

```diff
@@ -0,0 +1,365 @@
+                realAdmin: realAdmin
+            })
+        );
+    }
+
+    function _configureWorkflow(
+        DeployedContracts memory deployedContracts,
+        DecodedConfigParams memory decodedConfigParams,
+        address beneficiary
+    ) private {
+        // get bonding curve / funding manager
+        address fundingManager =
+            address(deployedContracts.orchestrator.fundingManager());
+        // get payment router
+        address[] memory modules = deployedContracts.orchestrator.listModules();
+        address paymentRouter;
+        bytes4 paymentRouterInterfaceId =
+            type(ILM_PC_PaymentRouter_v1).interfaceId;
+        for (uint i; i < modules.length; ++i) {
+            if (
+                ERC165(modules[i]).supportsInterface(
+                    paymentRouterInterfaceId
+                )
+            ) {
+                paymentRouter = modules[i];
+            }
+        }
+
+        // enable bonding curve to mint issuance token
+        deployedContracts.mintWrapper.setMinter(fundingManager, true);
+
+        // transfer initial collateral supply from realAdmin to bonding curve and mint issuance token to msg.sender
+        _manageInitialSupplies(
+            IBondingCurveBase_v1(fundingManager),
+            decodedConfigParams.collateralToken,
+            deployedContracts.issuanceToken,
+            decodedConfigParams.initialCollateralSupply,
+            decodedConfigParams.initialIssuanceSupply,
+            decodedConfigParams.realAdmin,
+            beneficiary
+        );
+
+        // assign permissions to buy/sell from curve to admin
+        bytes32 curveAccess = FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1(
+            fundingManager
+        ).CURVE_INTERACTION_ROLE();
+        FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1(fundingManager)
+            .grantModuleRole(curveAccess, beneficiary);
+
+        // assign payment pusher role to beneficiary
+        bytes32 paymentPusherRole =
+            ILM_PC_PaymentRouter_v1(paymentRouter).PAYMENT_PUSHER_ROLE();
+        IModule_v1(paymentRouter).grantModuleRole(
+            paymentPusherRole, beneficiary
+        );
+
+        // revoke privileges from factory
+        _renounceTokenPrivileges(deployedContracts.issuanceToken);
+        _transferWorkflowAdminRights(
+            deployedContracts.orchestrator, decodedConfigParams.realAdmin
+        );
+        _transferOwnership(
+            address(deployedContracts.mintWrapper),
+            decodedConfigParams.realAdmin
+        );
+    }
+
+    function _manageInitialSupplies(
+        IBondingCurveBase_v1 fundingManager,
+        IERC20 collateralToken,
+        ERC20Issuance_v1 issuanceToken,
+        uint initialCollateralSupply,
+        uint initialIssuanceSupply,
+        address admin,
+        address beneficiary
+    ) private {
+        uint availableFunding =
+            fundings[admin][beneficiary][address(collateralToken)];
```

Comment:
Anyone can execute the deployment for a given admin/beneficiary pair, but the beneficiary will always be the recipient of initial issuance token supply and CURVE_INTERACTION_ROLE on the newly deployed workflow, which in turn receives the initialCollateralSupply. So if Alice chooses to sponsor the bonding curve deployment for Bob, the factory makes sure that if anyone executes the deployment, Alice remains the admin of the workflow and Bob receives initial issuance supply as well as aforementioned role.

So I don't see how funds can be stolen, or am I missing something? :)

I renamed `actor` (which has been relict of previous logic) to `beneficiary` to make this clearer.
@omega-audits

Reply:
It doesn't matter who is the recipient of the tokens, since anyone can control the entire configs and modules added. For example anyone can deploy with a module transferring all the funds to themselves, or set themselves to control the authorizer etc. The tokens issues are then just worthless because the funds of the organization will be stolen.

Reply:
So the scenario would be this:

- target of the attack is the collateralToken that is in the factory, and specifically the amount registered here `    fundings[admin][beneficiary][address(collateralToken)];` for some admin, beneficiary 
- attacker calls createPIMWorkflow with arguments a follows:
   - adin adn benificiary as in the target
   - a fundingManager configuration that allows the attacker to access the funds in teh funding manager
   - `initialCollateralSupply` equal to `fundings[admin][beneficiary][address(collateralToken)]`
- this call will transfer initialCollateralSupply collateralToken to the funding manager, where the attacker can take it

I really should write a test for this, so I may be missing something essential here

Reply:
>  - a fundingManager configuration that allows the attacker to access the funds in teh funding manager
> initialCollateralSupply equal to fundings[admin][beneficiary][address(collateralToken)]

Hmm, but isn't is so that for being able to steal funds you need to be the orchestrator admin? And the factory makes sure that the admin is going to be the one who sponsors the whole bonding curve deployment?

Phrased differently: the factory makes sure who is the orchestrator admin of a workflow deployment (namely the address that added the funding in the first place). So only the address that added the funding could steal it. The only other privileges that are handed out through the factory are the CURVE_INTERACTION_ROLE and the PAYMENT_PUSHER_ROLE for the beneficiary. So only the beneficiary could potentially use the PAYMENT_PUSHER_ROLE to drain the reserve, but it is determined by the funder and is represented by a multisig which requires the funder's consent for transactions.

Reply:
I realize something else though:

While the factory controls who becomes orchestrator admin and who gets the payment pusher role and therefore makes sure that nobody else can steal funds there is a possibility for a malicious person to mess workflows up. So someone could come and deploy a `Simple Payment Processor` instead of a `Streaming Payment Processor` (as is actually intended), so that the workflow would need to be reconfigured by the orchestrator admin to fulfill its original purpose (which involves streaming). 

So I think it makes sense to check in the factory that the expected modules are actually deployed and not some other modules. This is related to one of your [previous comments](https://github.com/InverterNetwork/contracts/pull/672#discussion_r1801118855) I guess.